### PR TITLE
add undici fetch for timeout over 5mins

### LINKED ISSUE #5236: `HeadersTimeoutError` occurs in timeout of more than 5 mins in `fetchAdapter`
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```typescript
const result = await fetchAdapter(
      {
        url: this.url,
        method: "POST",
        data: JSON.stringify(request),
        headers: {
          "Content-Type":
            this.responseType === "stream"
              ? "text/event-stream"
              : "application/json",
        },
        responseType: this.responseType || "json",
        timeout: this.timeout,
      }
    ).then((res) => (this.responseType === "stream" ? res : res.data));
```

### Error Message and Stack Trace (if applicable)

```
{
    "name": "HeadersTimeoutError",
    "code": "UND_ERR_HEADERS_TIMEOUT",
    "message": "Headers Timeout Error"
}
```

### Description

`HeadersTimeoutError` occurs when I use timeout over 5 mins in `fetchAdapter`

My azure GPT api sometimes goes to 10 mins while the `HeadersTimeoutError` breaks the request at the 5th minute every time when I use the inner `fetchAdapter`

A related report is [https://github.com/nodejs/node/issues/46375](url)

### System Info

Mac, Linux
node v18.17.1
yarn 3.4.1

### PR DESCRIPTION
**Pull request by arthur19312**
Fixes #5236 
Use undici `agent` to pass on the timeout of `fetch` request, and it finally resolved the `HeadersTimeoutError` which just appears every time when the request go over 5mins.

### CODE REVIEW

File: langchain/src/util/axios-fetch-adapter.js [hunk 1]
```diff
@@ -230,10 +232,10 @@ export default async function fetchAdapter(config) {
  * Fetch API stage two is to get response body. This funtion tries to retrieve
```
[the remark this answers is not in the document]
Comment (arthur19312) [reply langchain/src/util/axios-fetch-adapter.js:232]: Yes, the node 18 inner `fetch` api cannot support timeout over 5mins by default signal, and more info could be found here [https://undici.nodejs.org/#/docs/api/Client?id=parameter-clientoptions](url)

**Review by bracesproul — COMMENTED** [review]
Ideally we aren't adding more deps, especially not deps which are over 1mb

Why would you want a 5 min timeout?

**Comment by arthur19312** [comment]
> 理想情况下，我们不会添加更多的 deps，尤其是超过 1mb 的 deps
> 
> 为什么你想要 5 分钟的超时？



> Ideally we aren't adding more deps, especially not deps which are over 1mb
> 
> Why would you want a 5 min timeout?

I know a sync request should be no more than several minutes, while I am just involved into the embarrasing situation that some of my Azure GPT requests take over 13mins, and the Azure api duration limit is in 15mins. Under most circumstances these requests should be asynchronous, however Azure apis haven't support it yet.

**Comment by bracesproul** [comment]
> > 理想情况下，我们不会添加更多的 deps，尤其是超过 1mb 的 deps
> > 为什么你想要 5 分钟的超时？
> 
> > Ideally we aren't adding more deps, especially not deps which are over 1mb
> > Why would you want a 5 min timeout?
> 
> I know a sync request should be no more than several minutes, while I am just involved into the embarrasing situation that some of my Azure GPT requests take over 13mins, and the Azure api duration limit is in 15mins. Under most circumstances these requests should be asynchronous, however Azure apis haven't support it yet.

At this moment we won't be able to merge a PR like this. We do have a `timeout` arg which most LLM/ChatModel classes accept. I would try that.

**Pull request closed without merging**